### PR TITLE
Nk refactor artifacts definition

### PR DESCRIPTION
**Pull request by vocrsz**
Problem: we need to copy 2 artifact-files from the same repository. Current artifact definition provides only one file to copy
Solution: change artifact definition syntax to provide multiple artifact-files configuration for the same repository

Before
```ruby
Indocker.add_artifact(
  Indocker::Artifacts::Git.new(
    name:        :artifact_file_1,
    remote_name: 'origin',
    remote_url:  'git@github.com:ArtStation/indocker.git',
    branch:      :master,
    source_path: 'Gemfile',
    target_path: File.join(Indocker.deploy_dir, 'assets/'),
  )
)

Indocker.add_artifact(
  Indocker::Artifacts::Git.new(
    name:        :artifact_file_2,
    remote_name: 'origin',
    remote_url:  'git@github.com:ArtStation/indocker.git',
    branch:      :master,
    source_path: 'Gemfile.lock',
    target_path: File.join(Indocker.deploy_dir, 'assets/'),
  )
)
```
After
```ruby
Indocker.add_artifact(
  Indocker::Artifacts::Git.new(
    name:        :artifact_file,
    remote_name: 'origin',
    remote_url:  'git@github.com:ArtStation/indocker.git',
    branch:      :master,
    files: [
      Indocker::Artifacts::DTO::FilesDTO.new(
        source_path: 'Gemfile',
        target_path: File.join(Indocker.deploy_dir, 'assets/')
      ),
      Indocker::Artifacts::DTO::FilesDTO.new(
        source_path: 'Gemfile.lock',
        target_path: File.join(Indocker.deploy_dir, 'assets/')
      ),
    ]
  )
)
```

### CODE REVIEW

File: lib/indocker/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Indocker
-  VERSION = "0.1.10"
+  VERSION = "0.1.13"
```
Comment (gvalmon) [inline lib/indocker/version.rb]: I think we should not have version change in PR

File: lib/indocker/artifacts/git.rb [hunk 2]
```diff
@@ -1,13 +1,12 @@
 class Indocker::Artifacts::Git
-  attr_reader :name, :remote_name, :remote_url, :branch, :source_path, :target_path
+  attr_reader :name, :remote_name, :remote_url, :branch, :files
 
-  def initialize(name:, remote_name:, remote_url:, branch:, source_path:, target_path:)
+  def initialize(name:, remote_name:, remote_url:, branch:, files:)
```
Comment (gvalmon) [inline lib/indocker/artifacts/git.rb]: I would keep the previous way as well, and mark it as deprecated, to improve the migration process.

Comment (gvalmon) [reply lib/indocker/artifacts/git.rb]: Maybe we can even do not mark it as deprecated and use both solutions.

File: lib/indocker/version.rb (hunk 1 above)
Comment (vocrsz) [reply lib/indocker/version.rb]: 👍

File: lib/indocker/artifacts/git.rb (hunk 2 above)
Comment (vocrsz) [reply lib/indocker/artifacts/git.rb]: 👍

File: lib/indocker/launchers/configuration_deployer.rb [hunk 3]
```diff
@@ -423,23 +423,45 @@ def sync_artifacts(clonner, artifact_servers)
               logger: @logger
             )
 
-            @logger.info("Pulling git artifact  #{artifact.name.to_s.green} for #{server.user}@#{server.host}")
-            result = clonner.clone(session, artifact.repository)
+            if artifact.is_a?(Indocker::Artifacts::Git)
```
Comment (gvalmon) [inline lib/indocker/launchers/configuration_deployer.rb]: we should not have `IF` condition for artifact in configuration deployer. ConfigurationDeployer is very high level class and should not know about different types of artifacts. Please move this to a separate class and try to refactor so we don't have `IF` condition.

**Pull request merged**